theme/src/components/GlobalTableOfContents.tsx

```diff
@@ -25,8 +25,53 @@ const HEADER_TITLES = [Header.Level2, Header.Level3];
  * data is deeply nested.
  */
 export function GlobalTableOfContents({ headers, rootHeaders }: Props) {
+  // Ref for the DOM node of the root unordered list element. This is
+  // used to get the maximum width of the container, which is used for
+  // determining if the current item overflows.
+  const listRef = useRef<HTMLOListElement>(null);
+
+  // Ref for the first DOM node that contains a list item. This node is used to
+  // measure the left border and padding to determine if the current item
+  // overflowed or not. Only one is needed because every item has the same left
+  // border and padding values.
+  const itemContainerRef = useRef<HTMLDivElement>();
+
+  /// Ref for an array of DOM nodes that correspond to a TOC item.
+  const itemsRef = useRef(new Map<string, HTMLElement>());
+
+  // Set containing list of strings that can be used to determine if a tooltip is enabled.
+  const [enabledTooltipSet, setEnabledTooltipSet] = useState(new Set<string>());
+
+  // Next.js router path.
   const currentPathname = useCurrentPathname();
 
+  // Effect that creates a new set of header IDs whose content is truncated so
+  // that we can conditionally render content tooltips.
+  useEffect(() => {
+    const nextSet = new Set<string>();
+
+    // Get left padding and border pixel values of the surrounding div container.
+    const computedStyle =
+      itemContainerRef.current &&
+      window.getComputedStyle(itemContainerRef.current);
+    const paddingLeft = parseFloat(computedStyle?.paddingLeft ?? '0');
+    const borderLeft = parseFloat(computedStyle?.borderLeft ?? '0');
+
+    for (const [headerId, node] of itemsRef.current.entries()) {
+      // Total item is the node width + the parent container's left border and padding.
+      const width = borderLeft + paddingLeft + node.offsetWidth;
+
+      // Check if current node overflows the parent list node:
+      // https://stackoverflow.com/a/10017343
```

Comment:
Just want to clarify if this stackoverflow link is purposed to be here for some context or not?

Reply:
yep this is for context on the approach I'm using 😄

Reply:
although, it's slightly modified because i'm also including the left border and padding in the total width

Reply:
got it!